### PR TITLE
feat(tailwind): create a tailwind.config.mjs file by default

### PR DESCRIPTION
**Pull request by florian-lefebvre**
## Changes

- Since Astro is ESM only and that Tailwind supports ESM, default config file is now `tailwind.config.mjs` (instead of `.cjs`).

## Testing

Not tested, I searched in the whole repo for `tailwind.config` occurences and updated relevant ones (eg. not in fixtures)

## Docs

Updated the file extension in the docs

### CODE REVIEW

**Review by sarah11918 — APPROVED** [review]
Looks fine to docs! :raised_hands:

File: packages/integrations/tailwind/README.md [hunk 1]
```diff
@@ -157,7 +157,7 @@ error   The `text-special` class does not exist. If `text-special` is a custom c
 [Instead of using `@layer` directives in a global stylesheet](https://tailwindcss.com/docs/functions-and-directives#using-apply-with-per-component-css), define your custom styles by adding a plugin to your Tailwind config to fix it:
 
 ```js
-// tailwind.config.cjs
+// tailwind.config.mjs
 module.exports = {
```
Comment (natemoo-re) [inline packages/integrations/tailwind/README.md:161]:
```suggestion
// tailwind.config.mjs
export default {
```

File: packages/astro/src/cli/add/index.ts [hunk 2]
```diff
@@ -160,7 +160,7 @@ export async function add(names: string[], { flags }: AddOptions) {
 						'./tailwind.config.mjs',
 						'./tailwind.config.js',
 					],
-					defaultConfigFile: './tailwind.config.cjs',
+					defaultConfigFile: './tailwind.config.mjs',
 					defaultConfigContent: TAILWIND_CONFIG_STUB,
```
Comment (natemoo-re) [inline packages/astro/src/cli/add/index.ts:164]: We also need to change this stub file to use ESM!

**Review by natemoo-re — DISMISSED** [review]
Excited about this!

File: packages/astro/src/cli/add/index.ts (hunk 2 above)
Comment (florian-lefebvre) [reply packages/astro/src/cli/add/index.ts:164]: Done!

**Review by bluwy — APPROVED** [review]

**Comment by delucis** [comment]
Noting that we should update Starlight’s docs to show `.mjs` and not `.cjs` once this is merged: https://starlight.astro.build/guides/css-and-tailwind/#tailwind-css

Potentially the same for the Fonts guide in Astro’s docs: https://docs.astro.build/en/guides/fonts/#register-fonts-in-tailwind (although that’s less directly impacted as it’s not showing Tailwind set-up in the same way).

**Review by ematipico — APPROVED** [review]

**Pull request merged**